setup.sh

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+#
+# Doubtfire MacOS Development Install Script
+# =======================
+#
+# Provides commands to install Doubtfire
+#
+
+#
+# OS Detection functions
+#
+isMac() {
+    if [[ `uname` == Darwin ]]; then
+        # 0 = true
+        return 0
+    else 
+        return 1
+    fi
+}
+
+isLinux() {
+    if [[ `uname` == Linux ]]; then
+        # 0 = true
+        return 0
+    else 
+        return 1
+    fi
+}
+
+is_zsh() {
+    if [ -n "$ZSH_VERSION" ]; then
+        # 0 = true
+        return 0
+    elif [ -n "$BASH_VERSION" ]; then
+        return 1
+    fi
+}
+
+#
+# Resets the color
+#
+msg_reset () {
+  RESET='\033[0m'
+  printf "${RESET}\n"
+}
+
+#
+# Log an error
+#
+error () {
+  RED_FORE='\033[0;31m'
+  printf "${RED_FORE}ERROR: $1"
+  msg_reset
+}
+
+#
+# Log verbose message
+#
+verbose () {
+  if [ $VERBOSE_OUTPUT -eq 1 ]; then
+    return
+  fi
+  CYAN_FORE='\033[0;36m'
+  printf "${CYAN_FORE}INFO: $1"
+  msg_reset
+}
+
+#
+# Log message
+#
+msg () {
+  printf "$1\n"
+}
+
+#
+# Install Homebrew
+#
+install_homebrew () {
+    msg "Installing Homebrew"
+
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+    brew tap caskroom/cask
+
+    if [ $? -ne 0 ]; then
+        error "Was not able to install Homebrew."
+        exit 1
+    fi
+}
+
+#
+# Install ruby envioronment for linux
+#
+install_rbenv_linux() {
+    msg "installing ruby"
+    sudo apt update
+    git clone git://github.com/sstephenson/rbenv.git ~/.rbenv
+    git clone git://github.com/sstephenson/ruby-build.git ~/.rbenv/plugins/ruby-build
+
+    sudo apt-get install -y libreadline-dev
+
+    msg "git repos cloned"
```

Comment:
Let's use capital sentence here -- "Git repos cloned". Could this be a `verbose` instead?